samples/snippets/src/test/java/com/example/spanner/GetCommitStatsSampleIT.java

```diff
@@ -0,0 +1,150 @@
+      if (iterator.hasNext()) {
+        instanceId = iterator.next().getId().getInstance();
+      }
+    }
+    dbId = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
+    dbClient.dropDatabase(dbId.getInstanceId().getInstance(), dbId.getDatabase());
+    dbClient
+        .createDatabase(
+            instanceId,
+            databaseId,
+            ImmutableList.of(
+                "CREATE TABLE Singers ("
+                    + "  SingerId   INT64 NOT NULL,"
+                    + "  FirstName  STRING(1024),"
+                    + "  LastName   STRING(1024),"
+                    + "  SingerInfo BYTES(MAX)"
+                    + ") PRIMARY KEY (SingerId)",
+                "CREATE TABLE Albums ("
+                    + "  SingerId        INT64 NOT NULL,"
+                    + "  AlbumId         INT64 NOT NULL,"
+                    + "  AlbumTitle      STRING(MAX),"
+                    + "  MarketingBudget INT64"
+                    + ") PRIMARY KEY (SingerId, AlbumId),"
+                    + "  INTERLEAVE IN PARENT Singers ON DELETE CASCADE",
+                "CREATE INDEX AlbumsByAlbumTitle ON Albums(AlbumTitle)"))
+        .get();
+  }
+
+  @AfterClass
+  public static void dropTestDatabase() {
+    dbClient.dropDatabase(dbId.getInstanceId().getInstance(), dbId.getDatabase());
+    spanner.close();
+  }
+
+  @Before
+  public void insertTestData() {
+    final DatabaseClient client = spanner.getDatabaseClient(dbId);
+    client.write(Arrays.asList(
+        Mutation.newInsertBuilder("Singers")
+            .set("SingerId")
+            .to(1L)
+            .set("FirstName")
+            .to("first name 1")
+            .set("LastName")
+            .to("last name 1")
+            .build(),
+        Mutation.newInsertBuilder("Singers")
+            .set("SingerId")
+            .to(2L)
+            .set("FirstName")
+            .to("first name 2")
+            .set("LastName")
+            .to("last name 2")
+            .build()
+    ));
+  }
+
+  @After
+  public void removeTestData() {
+    final DatabaseClient client = spanner.getDatabaseClient(dbId);
+    client.write(Collections.singletonList(Mutation.delete("Singers", KeySet.all())));
+  }
+
+  @Test
+  public void testGetCommitStatsSample() {
+    final DatabaseClient client = spanner.getDatabaseClient(dbId);
+    final String out = runExample(client);
+
+    assertThat(out).contains("Updated data with 8 mutations.");
+  }
+
+  private String runExample(DatabaseClient client) {
+    PrintStream stdOut = System.out;
+    ByteArrayOutputStream bout = new ByteArrayOutputStream();
+    PrintStream out = new PrintStream(bout);
+    System.setOut(out);
+    GetCommitStatsSample.getCommitStats(client);
+    System.setOut(stdOut);
+    return bout.toString();
+  }
+
+  private static String formatForTest(String name) {
+    return name + "-" + UUID.randomUUID().toString().substring(0, 20);
```

Comment:
This risks creating a database name that is longer than 30 characters. The default name for the test database is `commitstatssample`, which would mean that on an environment where the `"spanner.sample.database"` environment variable is not set, the name would become too long.

Reply:
Fixed.